packages/world-modules/src/modules/erc20-puppet/ERC20Module.sol

```diff
@@ -80,6 +81,7 @@ contract ERC20ModuleRegistrationLibrary {
     // Register the tables
     Allowances.register(_allowancesTableId(namespace));
     Balances.register(_balancesTableId(namespace));
+    TotalSupply.register(_totalSupplyTableId(namespace));
```

Comment:
are there any strategies we have available to keep this sort of thing from happening again? ideally we don't want to add runtime checks (for extra gas in hot path) and tests have a similar "manual maintenance" problem

Reply:
This is non-trivial because there is no "list" of all the tables that are registered by a module - the table registration is just inlined in some library. 

We sort of have to manually maintain this, but we can make it a bit more sticky by "bundling" it in with other tests. Like we can adopt a pattern of "whenever we test reading a table value, we check it exists first"

Reply:
Although in this case, we never test `TotalSupply.get()`, we only ever read it via `totalSupply()` 🤔

Reply:
are there other kinds of tests we can do? if each module had its own mud config, we could potentially have a test that reads the config and checks that each table has been registered during deploy?

Reply:
We could have separate MUD configs for each config, although even that wouldn't cover every case. For example `KeysWithValue` uses `tableIdArgument` so you'd have to figure out the target table ID.

Reply:
just to clarify, we don't have to solve this in this PR but just wanna think about how we can prevent these sorts of things falling through the cracks in the future

Reply:
Maybe an end to end test that throws if a store event is retrieved but no corresponding table registration event? (Would only have captured this case if we used the ERC20 module in an end-to-end test though...)

Reply:
ohh! I think there's a way to track storage reads with forge-std, so we could have a general test to detect storage reads and ensure that the associated table is registered before reading

Reply:
added a follow up here: https://github.com/latticexyz/mud/issues/2883